IPython/core/completer.py

```diff
@@ -472,6 +472,18 @@ def _safe_isinstance(obj, module, class_name):
     return (module in sys.modules and
             isinstance(obj, getattr(__import__(module), class_name)))
 
+def _safe_really_hasattr(obj, name):
```

Comment:
You're right, we do already have such a canary. I don't think `safe_hasattr` is quite the right function for it, but I'll try to find a way to combine them.